packages/aws-cdk-lib/aws-cloudwatch/lib/variable.ts

```diff
@@ -0,0 +1,231 @@
+   */
+  public static fromSearch(expression: string, populateFrom: string): Values {
+    return new SearchValues(expression, populateFrom);
+  }
+
+  /**
+   * Create values from an array of possible variable values
+   */
+  public static fromValues(...values: VariableValue[]): Values {
+    if (values.length == 0) {
+      throw new Error('Empty values is not allowed');
+    }
+    return new StaticValues(values);
+  }
+
+  public abstract toJson(): any;
+}
+
+class StaticValues extends Values {
+  constructor(private readonly values: VariableValue[]) {
+    super();
+  }
+
+  toJson(): any {
+    return {
+      values: this.values.map(value => ({ label: value.label, value: value.value })),
+    };
+  }
+}
+
+class SearchValues extends Values {
+  /**
+   * Create a search expression for use
+   *
+   * @param expression search expression that specifies a namespace, dimension name(s) and a metric name. For example `{AWS/EC2,InstanceId} MetricName=\"CPUUtilization\"`
+   * @param populateFrom dimension the dimension name, that the search expression retrieves, whose values will be used to populate the values to choose from. For example `InstanceId`
+   */
+  public constructor(public readonly expression: string, public readonly populateFrom: string) {
+    super();
+  }
+
+  toJson(): any {
+    return {
+      search: this.expression,
+      populateFrom: this.populateFrom,
+    };
+  }
+}
+
+/**
+ * Default value for use in {@link DashboardVariableOptions}
+ */
+export class DefaultValue {
+
+  /**
+   * A special value for use with search expressions to have the default value be the first value returned from search
+   */
+  public static readonly FIRST = new DefaultValue('__FIRST');
+
+  /**
+   * Create a default value
+   * @param value the value to be used as default
+   */
+  public static of(value: any) {
+    return new DefaultValue(value);
+  }
+
+  private constructor(public readonly value: any) { }
+}
+
+/**
+ * Options for {@link DashboardVariable}
+ */
+export interface DashboardVariableOptions {
+  /**
+   * Type of the variable
+   */
+  readonly type: VariableType;
+
+  /**
+   * The way the variable value is selected
+   */
+  readonly inputType: VariableInputType;
+
+  /**
+   * Pattern or property value to replace
+   */
+  readonly value: string;
+
+  /**
+   * Unique id
+   */
+  readonly id: string;
+
+  /**
+   * Optional label in the toolbar
+   *
+   * @default - the variable's value
+   */
+  readonly label?: string;
+
+  /**
+   * Optional values (required for {@link VariableInputType.RADIO} and {@link VariableInputType.SELECT} dashboard variables).
+   *
+   * @default - no values
+   */
+  readonly values?: Values;
+
+  /**
+   * Optional default value
+   *
+   * @default - no default value is set
+   */
+  readonly defaultValue?: DefaultValue;
```

Comment:
I feel like the word value is oversaturated in this PR... would it make more sense to name this `default`? Or do you want to keep `value`, `values`, and `defaultValue`?

Reply:
In this one, I kinda feel `defaultValue` might be the clearest!